src/ProductList.js

```diff
@@ -45,35 +48,55 @@ const filterByCategory = (setActiveFilters, activeFilters, id) => {
   }
 };
 
-const ProductList = ({ products, filterTypes }) => {
-  const [activeFilters, setActiveFilters] = useState([]);
+const ProductList = () => {
+  const [searchParams] = useSearchParams();
+  const categoryFilter = searchParams.get("category");
 
-  let filteredProducts = {
-    results:
-      activeFilters.length === 0
-        ? products.results
-        : products.results.filter((product) =>
-            activeFilters.includes(product.data.category.id)
-          ),
-  };
+  const [activeFilters, setActiveFilters] = useState(
+    categoryFilter !== null ? [categoryFilter] : []
+  );
+
+  const { data: productsData, isLoading: isLoadingProducts } = useProducts();
+
+  const { data: categoriesData, isLoading: isLoadingCategories } =
+    useProductCategories();
+
+  let filteredProducts = [];
+
+  if (isLoadingProducts === false) {
+    filteredProducts = {
+      results:
+        activeFilters.length === 0
+          ? productsData.results
+          : productsData.results.filter((product) =>
+              activeFilters.includes(product.data.category.id)
+            ),
+    };
+  }
 
   return (
     <>
       <Sidebar>
         <FilterTittle>Filters</FilterTittle>
         <div>
-          {filterTypes.map((category) => (
-            <div key={category.id}>
-              <FilterButton
-                onClick={() =>
-                  filterByCategory(setActiveFilters, activeFilters, category.id)
-                }
-                active={activeFilters.includes(category.id)}
-              >
-                {category.data.name}
-              </FilterButton>
-            </div>
-          ))}
+          {isLoadingCategories
+            ? null
+            : categoriesData.results.map((category) => (
+                <div key={category.id}>
+                  <FilterButton
+                    onClick={() =>
+                      filterByCategory(
+                        setActiveFilters,
+                        activeFilters,
+                        category.id
+                      )
+                    }
+                    active={activeFilters.includes(category.id)}
+                  >
+                    {category.data.name}
+                  </FilterButton>
+                </div>
+              ))}
```

Comment:
Esto puede ser su propio componente también, involucra más lógica de render que realmente a este componente padre no le importa. Las únicas dependencias que veo son, `categoriesData` y `activeFilters`.

Por último, podemos hacer la validación más corta con la negación de `isLoadingCategories`y el operador `&&`, así:

```suggestion
          { !isLoadingCategories &&
            : categoriesData.results.map((category) => (
                <div key={category.id}>
                  <FilterButton
                    onClick={() =>
                      filterByCategory(
                        setActiveFilters,
                        activeFilters,
                        category.id
                      )
                    }
                    active={activeFilters.includes(category.id)}
                  >
                    {category.data.name}
                  </FilterButton>
                </div>
              ))}
```